### PR TITLE
[Weather] Durability loss application

### PR DESCRIPTION
**Pull request by Unarelith**

### CODE REVIEW

**Comment by Mystos** [comment]
Task linked: [CU-ff4zhq Durability Cloth is not used and not setup](https://app.clickup.com/t/ff4zhq)

**Comment by Unarelith** [comment]
Ready for review

**Review by GrandPiaf — CHANGES_REQUESTED** [review]
"Assets/New Terrain.asset" ? Should be renamed and moved to correct folder.

File: Snowy-Path/Assets/Scripts/Character/Environment/WeatherPreset.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(menuName = "Environment/Weather")]
+public class WeatherPreset : ScriptableObject
+{
+    [System.Serializable]
+    public enum EWeatherType {
```
Comment (GrandPiaf) [inline Snowy-Path/Assets/Scripts/Character/Environment/WeatherPreset.cs]: Pleaser remove useless EWeatherType.

**Review by GrandPiaf — CHANGES_REQUESTED** [review]

File: Snowy-Path/Assets/Scripts/Character/Temperature.cs [hunk 2]
```diff
@@ -128,7 +128,7 @@ float GetTemperatureLossRate()
         if (cloth != null)
             clothType = (float)cloth.type;
 
-        return (1 + m_weather.GetCurrentWeather().blizzardStrength) - ((1 + m_weather.GetCurrentWeather().blizzardStrength) * clothType / 100f);
+        return (1 + (float)m_weather.GetCurrentWeather().blizzardStrength / 100f) - ((1 + (float)m_weather.GetCurrentWeather().blizzardStrength / 100f) * clothType / 100f);
```
Comment (GrandPiaf) [inline Snowy-Path/Assets/Scripts/Character/Temperature.cs:131]: Could be better to cache GetCurrentWeather() return value.

Comment (Unarelith) [reply Snowy-Path/Assets/Scripts/Character/Temperature.cs:131]: Automatically cached by your Weather version

File: Snowy-Path/Assets/Scripts/Character/Environment/WeatherPreset.cs (hunk 1 above)
Comment (Unarelith) [reply Snowy-Path/Assets/Scripts/Character/Environment/WeatherPreset.cs]: Done

**Comment by Unarelith** [comment]
Branch is ready

**Review by GrandPiaf — APPROVED** [review]

**Pull request merged**